pkg/driverbuilder/builder/ubuntu.go

```diff
@@ -251,43 +167,26 @@ func fetchUbuntuGenericKernelURL(baseURL string, kr kernelrelease.KernelRelease,
 			kr.Architecture.String(),
 		),
 		fmt.Sprintf(
-			"%s/linux-headers-%s%s-generic_%s-%s.%d_%s.deb",
-			baseURL,
+			"linux-headers-%s-%s-%s_%s-%s.%s_%s.deb",
```

Comment:
During a Ubuntu Azure test, the second and the last pattern that you check were identical, and due to this, the function `getResolvingURLs` return a array with 3 validated URLs, eventually the function `ubuntuHeadersURLFromRelease` return an error because it was expecting exactly 2 values.
Differents ways to fix it:
- Change the `len(urls) == 2` to `len(urls) >= 2` in `ubuntuHeadersURLFromRelease` function
- Checking for duplicated before returning the patterns in `fetchUbuntuKernelURL` function

Config:
```
kernelversion: 46
kernelrelease: 5.4.0-1044-azure
target: ubuntu
architecture: amd64
output:
  module: output/39ae7d40496793cf3d3e7890c9bbdc202263836b/falco_ubuntu_5.4.0-1044-azure_46.ko
  probe: output/39ae7d40496793cf3d3e7890c9bbdc202263836b/falco_ubuntu_5.4.0-1044-azure_46.o
```

Reply:
This will happen every time the extra version is being correctly parsed by the function `parseUbuntuExtraVersion` because you rebuild the extra version by splitting it with a dash as you parsed it.

Reply:
Good catch! Let me work up a fix.

Reply:
I went for just deduplicating the URL slice. In the worst case scenario, `parseUbuntuExtraVersion` just passes back to the user exactly what the user passed in since there is no error clause. In that case, I'd like to keep as many possible naming patterns and check for duplicates after the fact.

But it's probably also just good practice to deduplicate the slice anyway. If we can ensure that we're only getting 2 URLs to resolve, then we are also reducing the amount of `curl` calls and downloads/bandwidth.

Reply:
There is an argument for just removing the bottom naming pattern altogether though. Since `parseUbuntuExtraVersion` has no error clause, it could be said that it will *always* parse the extra version correctly, and therefore always cause duplicates.

Reply:
Yea, all good 😄 I know what you mean - and I think leaving it in there is fine in the event that the function is unable to split on `-` and just returns back whatever was passed in. I'm weary there may be some edge case in that scenario. And there should be no harm leaving it in with a `dedup` function cleaning it up.